### ISSUE TITLE
GPU is not correct in UI

### ISSUE THREAD

**Issue by junmingzha** [gpustack/gpustack#212]
**Describe the bug**

GPU is not correct in UI. It is not nvidia 4080 in nvidia-smi  command. and also no error log in log.
**Steps to reproduce**

**Result**

**Expected behavior**

**Environment**

- GPUStack version: 0.1.2
- OS: ubuntu
- GPU: nvidia 4080

**Additional context**

![image](https://github.com/user-attachments/assets/77fda024-d5ed-43ad-ab26-b45da4192251)
![image](https://github.com/user-attachments/assets/c76422ed-c996-4873-a132-fdee4c239a99)

**Comment by pengjiang80** [comment]
Is it a standard physical machine or a virtual machine with GPU passthrough or other things? We have some similar issues reported in virtual machines with GPU passthrough.

**Comment by aiwantaozi** [comment]
Hi, thank you for your feedback. 

We have a preliminary script for the Linux x86 architecture to help check for problems during the GPU detection. Please run the following command:

```bash
sudo ./detect_gpus.sh >> detect_gpus.log
```

After running the command, check if the `detect_gpus.log` contains any "Found GPU" logs and whether the GPU part in `fastfetch_output.log` can detect the GPU correctly. Please upload the `detect_gpus.log`, `fastfetch_output.log`, and `/var/log/gpustack.log` to the issue so that we can jointly review if our GPU detection has overlooked any scenarios relevant to your environment. 

[detect_gpus.sh.zip](https://github.com/user-attachments/files/16609198/detect_gpus.sh.zip)

If your architecture is not Linux x86, please let us know your environment details, we will try to simulate your environment to resolve the issue.

**Comment by junmingzha** [comment]
Attached are log files after that executed your script. Please check it.
[gpustack.log](https://github.com/user-attachments/files/16861435/gpustack.log)
[detect_gpus.log](https://github.com/user-attachments/files/16861436/detect_gpus.log)
[fastfetch_output.log](https://github.com/user-attachments/files/16861437/fastfetch_output.log)

**Comment by aiwantaozi** [comment]
Here’s a more polished and professional version of your message:

---

Thank you for your response. We have reviewed the logs you provided. We retrieves GPU device information from the Direct Rendering Manager (DRM) first. If GPU device information is successfully obtained, it uses that GPUs. If no GPU information is retrieved from DRM, it will then attempt to acquire the GPU device from the PCI interface.

Based on the logs, your environment contains both Aspeed and NVIDIA GPUs. The issue arises because the DRM can retrieve the Aspeed GPU but fails to detect the NVIDIA GPU. This could be due to the `nvidia_drm` kernel module not being installed or not being loaded.

The default NVIDIA CUDA installation will install and load the `nvidia_drm` kernel module. However, certain custom installation parameters can prevent `nvidia_drm` from being installed, such as using `--no-drm`. For more information, please refer to the documentation at [link](https://docs.nvidia.com/cuda/archive/12.5.1/cuda-installation-guide-linux/#advanced-options).

We recommend the following steps to workaround. Our upcoming version 0.2.0 will enhance the PCI detection process, providing a more comprehensive solution. Alternatively, you may choose to wait for this update.

### Suggested Workaround:
```bash
# Check if nvidia-drm is present on the system; if so, proceed with the following operations.
modinfo nvidia_drm

# Verify if the module is loaded, if it doesn't load, try to check why.
lsmod | grep nvidia_drm

# Check for blacklisting or if modeset is set to 0.
# If it is set to 0, change it to 1 by adding `options nvidia-drm modeset=1`.
# If it is blacklisted, remove it from the blacklist.
grep nvidia /etc/modprobe.d/*

# If you have made changes, update.
sudo update-initramfs -u

# Load the nvidia_drm module.
modprobe nvidia_drm

# Reboot the system.
reboot

# After rebooting, confirm that nvidia-drm has been loaded.
lsmod | grep nvidia_drm

# Restart GPUStack.
```

If `nvidia_drm` was not installed during the CUDA installation, you will need to install nvidia-drm to workaround it or wait for the optimizations in version 0.2.0.
```bash
# Install without specifying --no-drm to ensure nvidia_drm is installed.
# Note: This command is a sample and may need to be adjusted for your environment.
sudo sh cuda_12.6.1_560.35.03_linux.run --driver --toolkit
```

Pre-try the 0.2.0 detection way,  could run below command with using the PCI detection method, if this way can detect the GPUs for your environment, the coming 0.2.0 should work for you, if it has the problem detecting your GPUs, please let us know.
```
# download fastfetch and double-check the GPU
wget https://github.com/gpustack/fastfetch/releases/download/2.18.1.5/fastfetch-linux-amd64.zip
unzip fastfetch-linux-amd64.zip
cd fastfetch--linux-amd64/usr/bin
./fastfetch --gpu-driver-specific true --gpu-temp true --format json --gpu-detection-method pci
```

**Comment by junmingzha** [comment]
It works after I reinstall cuda. thanks a lot. but how to remove aspeed device?
![image](https://github.com/user-attachments/assets/5fd0c91b-8aec-494d-a7de-0d23d737e2ce)

**Comment by aiwantaozi** [comment]
Hi, this version currently doesn’t allow the removal of the integrated GPU. However, the integrated GPU won't be used for model deployment. The issue you report is quite helpful, we've removed the integrated GPU in the upcoming release, and you’ll be able to upgrade once it’s available. Thank you for your patience!

**Comment by junmingzha** [comment]
![image](https://github.com/user-attachments/assets/9e7909ff-401b-44a6-b35a-0bd14ddea11a)
root@aiserver:~# /usr/local/cuda-12.6/bin/nvcc
nvcc fatal   : No input files specified; use option --help for more information
root@aiserver:~# /usr/local/cuda-12.6/bin/nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2024 NVIDIA Corporation
Built on Fri_Jun_14_16:34:21_PDT_2024
Cuda compilation tools, release 12.6, V12.6.20
Build cuda_12.6.r12.6/compiler.34431801_0

**Comment by junmingzha** [comment]
after reinstalling cuda， the nvidia GPU can not be got in web UI.

**Comment by Finenyaco** [comment]
> ![image](https://private-user-images.githubusercontent.com/28426742/365909206-9e7909ff-401b-44a6-b35a-0bd14ddea11a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5NjYyNDQsIm5iZiI6MTcyNTk2NTk0NCwicGF0aCI6Ii8yODQyNjc0Mi8zNjU5MDkyMDYtOWU3OTA5ZmYtNDAxYi00NGE2LWIzNWEtMGJkMTRkZGVhMTFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEwVDEwNTkwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ5OGQ4ZDUxZTEwYTQyMzcxN2E2NTUxN2ZiMjMzOWEyNzc3ZWU1MTNlZDE4MzRlNTIwMjg5MTRmMGFlNGIxZGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-yuuaPLzA__L8aYprNszcYm3pK3BUV-EGsPvhsDCTCM) root@aiserver:~# /usr/local/cuda-12.6/bin/nvcc nvcc fatal : No input files specified; use option --help for more information root@aiserver:~# /usr/local/cuda-12.6/bin/nvcc --version nvcc: NVIDIA (R) Cuda compiler driver Copyright (c) 2005-2024 NVIDIA Corporation Built on Fri_Jun_14_16:34:21_PDT_2024 Cuda compilation tools, release 12.6, V12.6.20 Build cuda_12.6.r12.6/compiler.34431801_0

Hi, junmingzha

In the case you mentioned above, the nvidia-drm module has been loaded correctly. The new issue you've encountered appears to be related to another issue https://github.com/gpustack/gpustack/issues/221. Both https://github.com/gpustack/gpustack/issues/221 and https://github.com/gpustack/gpustack/issues/212 will be resolved in the upcoming version 0.2.0, please feel free to try it out again.

**Comment by Finenyaco** [comment]
**Environment**
- GPUStack version: main(5d45649)
- OS: Ubuntu 22.04
- GPU: NVIDIA GeForce RTX 4080

**Test Result:** pass


- Remove the nvidia-drm module.

```
(base) root@passthrough:~# sudo lsmod | grep nvidia_drm
(base) root@passthrough:~#
(base) root@passthrough:~# ls -hrlt /sys/class/drm
total 0
lrwxrwxrwx 1 root root    0 Sep 10 10:11 card0 -> ../../devices/pci0000:00/0000:00:01.0/drm/card0
-r--r--r-- 1 root root 4.0K Sep 10 10:11 version
lrwxrwxrwx 1 root root    0 Sep 10 10:11 card0-Virtual-4 -> ../../devices/pci0000:00/0000:00:01.0/drm/card0/card0-Virtual-4
lrwxrwxrwx 1 root root    0 Sep 10 10:11 card0-Virtual-3 -> ../../devices/pci0000:00/0000:00:01.0/drm/card0/card0-Virtual-3
lrwxrwxrwx 1 root root    0 Sep 10 10:11 card0-Virtual-2 -> ../../devices/pci0000:00/0000:00:01.0/drm/card0/card0-Virtual-2
lrwxrwxrwx 1 root root    0 Sep 10 10:11 card0-Virtual-1 -> ../../devices/pci0000:00/0000:00:01.0/drm/card0/card0-Virtual-1
```

- fastfetch cannot recognize the NVIDIA GPU.
```
{
    "type": "GPU",
    "result": [
      {
        "index": null,
        "uuid": "",
        "coreCount": null,
        "memory": {
          "dedicated": {
            "total": null,
            "used": null
          },
          "shared": {
            "total": null,
            "used": null
          }
        },
        "driver": "qxl",
        "name": "QXL paravirtual graphic card",
        "temperature": null,
        "type": "Unknown",
        "vendor": "Red Hat, Inc.",
        "platformApi": "DRM (card0)",
        "frequency": null,
        "deviceId": 1024,
        "coreUtilizationRate": null
      }
    ]
  },
```

```
(base) root@passthrough:~# nvidia-smi
Tue Sep 10 12:08:31 2024
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 555.58.02              Driver Version: 555.58.02      CUDA Version: 12.5     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 4080 ...    Off |   00000000:07:00.0 Off |                  N/A |
|  0%   33C    P8              8W /  320W |    3084MiB /  16376MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A      5124      C   ...box/llama-box-linux-amd64-cuda-12.5        246MiB |
|    0   N/A  N/A      5429      C   ...box/llama-box-linux-amd64-cuda-12.5       2824MiB |
+-----------------------------------------------------------------------------------------+
```

- Install GPUStack. The iGPU is being ignored.
<img width="1580" alt="image" src="https://github.com/user-attachments/assets/79075fa5-442d-499e-93d6-6f61c92b74ba">

- Deploy a model instance.
<img width="1497" alt="image" src="https://github.com/user-attachments/assets/bcc82c7f-e8bc-4936-a766-3617fa441fe0">


Notes: Closed as resolved. If this issue can be reproduced, please re-open it or file a new issue.
